knowledge_graph/network_class.py

```diff
@@ -64,35 +51,32 @@ def add_child_to_result(self, child, parent, edge_type):
             edge_type: The relationship between child and parent
                         i.e. causes, inhibits, etc
         """
-        self.result.append((parent.label[0], child.label[0], edge_type))
+        self.edge_triplets.append((parent.label[0], child.label[0], edge_type))
         if child not in self.visited:
             self.visited.add(child)
             for obj_prop in self.obj_properties:
-                self.node_family.append(
-                    (child, eval("iter(child." + obj_prop + ")"), obj_prop)
-                )
+                val = getattr(child, obj_prop)
+                rec = (child, iter(val), obj_prop)
+                self.node_family.append(rec)
 
-    def add_class_to_explore(self, class_name):
+    def add_class_to_explore(self, owl_class_obj: owlready2.entity.ThingClass):
         """Adds all nodes related to a particular class. Some of these nodes
         will not actually be a class, but that is irrelevant as they will get ignored.
 
             Parameters
             ----------
-            class_name: A node in the ontology
+            owl_class_obj: A node in the ontology
         """
         for obj_prop in self.obj_properties:
-            try:
-                self.class_family.append(
-                    (class_name, eval("iter(class_name." + obj_prop + ")"), obj_prop)
-                )
-            except:
-                pass
-        try:
-            self.class_family.append(
-                (class_name, iter(self.ontology.get_parents_of(class_name)), "is_a")
-            )  # the class(es) of the ont_class. This could pull classes that are just Restriction classes, so really should add code here that checks the class is found in self.ontology.classes() before adding it to the class_family.
-        except:
-            pass
+            if hasattr(owl_class_obj, obj_prop):
+                val = getattr(owl_class_obj, obj_prop)
+                rec = (owl_class_obj, iter(val), obj_prop)  # why iter()?
+                self.class_family.append(rec)
+
+        parents = self.ontology.get_parents_of(owl_class_obj)
```

Comment:
We had a try in the original code. I like the if hasattr above. Not sure if an if is necessary here as well.

Reply:
Andrew wants to know why you did not replace the try with an if here for this line. He thinks it might the try should be replaced with an if for this line too.